### PR TITLE
Fixing the float/double parsing logic to handle exponents that are too large/small

### PR DESCRIPTION
**Pull request by tannergooding**
This resolves https://github.com/dotnet/corefx/issues/35780

### CODE REVIEW

**Comment by tannergooding** [comment]
CC. @stephentoub, @danmosemsft 

The Roslyn RealParser and ES6100M test validation suite both pass under Release. I am also running these under Debug bits but that will take overnight to run.

Will be adding a regression test to CoreFX for this scenario.

File: src/System.Private.CoreLib/shared/System/Number.Parsing.cs [hunk 1]
```diff
@@ -32,6 +32,15 @@ internal partial class Number
         private const int Int64Precision = 19;
         private const int UInt64Precision = 20;
 
+        private const int DoubleMaxExponent = 309;
+        private const int DoubleMinExponent = -324;
+
+        private const int FloatingPointMaxExponent = DoubleMaxExponent;
+        private const int FloatingPointMinExponent = DoubleMinExponent;
```
Comment (stephentoub) [inline src/System.Private.CoreLib/shared/System/Number.Parsing.cs]: Why do we need these rather than just using DoubleMax/MinExponent?

Comment (tannergooding) [reply src/System.Private.CoreLib/shared/System/Number.Parsing.cs]: It makes it easier to expand to support things like `Binary128`, if that ever happens.

Comment (stephentoub) [reply src/System.Private.CoreLib/shared/System/Number.Parsing.cs]: Ok.

**Comment by tannergooding** [comment]
>  I am also running these under Debug bits but that will take overnight to run.

These finished without failure as well (and would previously assert all over the place for the given failure case and similar strings).

**Pull request merged**